security/DeputyGuardianModule-Proxy.md

```diff
@@ -0,0 +1,63 @@
+# Proxify the `DeputyGuardianModule`
+
+## Context
+
+![image](https://github.com/user-attachments/assets/466c7d28-f4d1-425d-82ec-eb1326287689)
+
+
+
+
+The `DeputyGuardianModule` is an important component for the Superchain security model. \
+The `DeputyGuardianModule`, also known as _DGM_, is a smart contract deployed on L1 (currently [here](https://etherscan.io/address/0xc6901f65369fc59fc1b4d6d6be7a2318ff38db5b)) that has, for example, the ability to widely pause the Superchain. \
+The `DGM` is owned by the _Foundation Operation Safe_ so only operation run from the _Foundation Operation Safe_ can be executed into the `DGM`. \
+To make sure we can widely pause the Superchain rapidly enough in case of an emergency, we need to presign the pause transaction from the signers of the _FoS_. \
+These transactions are also known as `PSP` for PreSigned Pause transactions. \
+This process requires a ceremony and this is slow and tedious to put in place.
+
+The list of available actions that the `DeputyGuardianModule` can execute are:
+| Function Name | Description |
+| ---------------------- | ---------------------------------------------------------- |
+| setRespectedGameType() | Change the current `gameType` used by the `OptimismPortal` |
+| blacklistDisputeGame() | Blacklist a disputeGame into the `OptimismPortal` |
+| unpause() | Unpause the withdrawal of the Superchain |
+| pause() | Pause the withdrawal of the Superchain |
+| setAnchorState()| Set a state into `AnchorStateRegistry` |
+
+
+## Problem Statement
+
+The existing `DeputyGuardianModule` is not _proxified_. \
+This is inconvenient each time we want to upgrade the `DeputyGuardianModule` to add a new feature or to fix a potential bug. \
+As this is not a proxy this require to redeploy the `DeputyGuardianModule` on L1. And also, to update the `Guardian` contract to add the new `DeputyGuardianModule` address as authorized module. \
+Moreover, changing the `DeputyGuardianModule` will break the current PSPs setup as these pause transactions are presigned with the **previous** `DeputyGuardianModule`. \
+Thus, we need to generate new PSPs through the tedious process of a ceremony to add the new `DeputyGuardianModule` for making the PSPs valid again. \
+We are clearly seeing that this is not a sustainable solution for the long term. 
+
+Additionally, we have to simulate the new PSPs and share them with other member of the Superchain.
+
+## Proposed Solution
+
+![image](https://github.com/user-attachments/assets/099258f4-19fb-44b8-84e9-670b78a2f868)
+
+Here we propose some modification of the architecture with a proxy contract on top of the `DeputyGuardianModule`, this will allow us to upgrade the `DeputyGuardianModule` and keeping the same address.
+Thus, this will not invalidate the PSPs and we will not need to regenerate new PSPs each time we upgrade the `DeputyGuardianModule`.
```

Comment:
It will invalidate the PSPs when we replace the DGM with the proxy though right? Perhaps we should wait until we need to upgrade the DGM for another reason, and then move to a proxy at the same time.

Reply:
> It will invalidate the PSPs when we replace the DGM with the proxy though right? 

This is correct. 
> Perhaps we should wait until we need to upgrade the DGM for another reason

Yes this is the idea! With [_Incident Response Improvements_](https://github.com/ethereum-optimism/design-docs/issues/120) We will do the upgrade during the improvements of our incident response SC on L1. 
I should write this down somewhere.